src/firefly/js/ui/InputAreaFieldView.jsx

```diff
@@ -0,0 +1,151 @@
+}
+
+const ICON_SPACE_STYLE= {
+    verticalAlign: 'middle',
+    paddingLeft: 3,
+    width: 16,
+    height: 16,
+    display:'inline-block'};
+
+
+export class InputAreaFieldView extends Component {
+    constructor(props) {
+        super(props);
+        this.warnIcon = null;
+        this.state = { hasFocus: false, infoPopup: false };
+    }
+
+    shouldComponentUpdate(np, ns) { return sCompare(this, np, ns); }
+
+
+    componentDidUpdate() {
+        var {infoPopup}= this.state;
+        if (infoPopup) {
+            var {warningOffsetX, warningOffsetY}= computeWarningXY(this.warnIcon);
+            var {message}= this.props;
+            this.hider = DialogRootContainer.showTmpPopup(makeInfoPopup(message, warningOffsetX, warningOffsetY));
+        }
+        else {
+            if (this.hider) {
+                this.hider();
+                this.hider = null;
+            }
+        }
+    }
+
+    makeWarningArea(warn) {
+        if (warn) {
+            return (
+                <div style={ICON_SPACE_STYLE}
+                     onMouseOver={() => this.setState({infoPopup:true})}
+                     onMouseLeave={() => this.setState({infoPopup:false})}>
+                    <img src={EXCLAMATION} ref={(c) => this.warnIcon= c}/>
+                </div>
+            );
+        }
+        else {
+            return <div style={ICON_SPACE_STYLE}/>;
+        }
+    }
+
+    render() {
+        var {hasFocus}= this.state;
+        var {visible,label,tooltip,rows,cols,labelWidth,value,style,wrapperStyle,
+             valid,size,onChange, onBlur, onKeyPress, showWarning, message, type}= this.props;
+        if (!visible) return null;
+        wrapperStyle = Object.assign({whiteSpace:'nowrap', display: this.props.inline?'inline-block':'block'}, wrapperStyle);
+        return (
+            <div style={wrapperStyle}>
+                {label && <InputFieldLabel label={label} tooltip={tooltip} labelWidth={labelWidth}/> }
+                <textarea style={Object.assign({display:'inline-block'}, style)}
+                          rows={rows}
+                          cols={cols}
+                          className={computeStyle(valid,hasFocus)}
+                       onChange={(ev) => onChange ? onChange(ev) : null}
+                       onFocus={ () => !hasFocus ? this.setState({hasFocus:true, infoPopup:false}) : ''}
+                       onBlur={ (ev) => {
+                                onBlur && onBlur(ev);
+                                this.setState({hasFocus:false, infoPopup:false});
+                            }}
+                       onKeyPress={(ev) => onKeyPress && onKeyPress(ev)}
+                       value={type==='file' ? undefined : value}
+                       title={ (!showWarning && !valid) ? message : tooltip}
+                       size={size}
+                       type={type}
+                />
+                {showWarning && this.makeWarningArea(!valid)}
+            </div>
+        );
+    }
+}
```

Comment:
no, is not unfortunately. I wanted to bring it as quick as possible to 
allow Polygon search but IMHO it needs to go into a different ticket 
because it is a full component not related to the panel itself that 
should be review on its own.

On 5/15/16 3:33 PM, Trey Roby wrote:
>
> In src/firefly/js/ui/InputAreaFieldView.jsx 
> <https://github.com/Caltech-IPAC/firefly/pull/85#discussion_r63299160>:
>
> > +                       onFocus={ () => !hasFocus ? this.setState({hasFocus:true, infoPopup:false}) : ''}
> > +                       onBlur={ (ev) => {
> > +                                onBlur && onBlur(ev);
> > +                                this.setState({hasFocus:false, infoPopup:false});
> > +                            }}
> > +                       onKeyPress={(ev) => onKeyPress && onKeyPress(ev)}
> > +                       value={type==='file' ? undefined : value}
> > +                       title={ (!showWarning && !valid) ? message : tooltip}
> > +                       size={size}
> > +                       type={type}
> > +                />
> > +                {showWarning && this.makeWarningArea(!valid)}
> > +            </div>
> > +        );
> > +    }
> > +}
>
> I can't make |InputAreaFieldView| show me an error. Is it hooked up yet.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/85/files/8bdfd737fd8631b45d10aff1ebb2742bee675e9e#r63299160>
>

-- 
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489